src/Angor/Client/Pages/Recover.razor

```diff
@@ -1173,34 +1221,58 @@
     {
         var selected = e.Value?.ToString();
 
-        if (selected != null)
+        if (selected != null && int.TryParse(selected, out int res) && res <= feeData.FeeEstimations.Fees.Count)
         {
-            if (int.TryParse(selected, out int res))
-            {
-                if (res <= feeData.FeeEstimations.Fees.Count)
-                {
-                    if (!passwordComponent.HasPassword())
-                    {
-                        showEndOfProjectModal = false;
-                        notificationComponent.ShowErrorMessage("Wallet password expired");
-                        return;
-                    }
+            // Set the selected position for UI
+            feeData.FeePosition = res;
+
+            // Get the fee estimation based on the slider position
+            var selectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
+
+            // Use the common method to recalculate the fee with the selected fee rate and confirmation blocks
+            await RecalculateEndOfProjectFee(selectedFeeEstimation.FeeRate, selectedFeeEstimation.Confirmations);
+        }
+    }
+
+    private async Task RecalculateEndOfProjectFee(decimal customFeeRate, int confirmations = 0)
+    {
+        if (!passwordComponent.HasPassword())
+        {
+            showEndOfProjectModal = false;
+            notificationComponent.ShowErrorMessage("Wallet password expired");
+            return;
+        }
 
-                    feeData.SelectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
+        try
+        {
+            var words = await passwordComponent.GetWalletAsync();
 
-                    var accountInfo = storage.GetAccountInfo(network.Name);
-                    var words = await passwordComponent.GetWalletAsync();
+            // Create a custom fee estimation with the provided rate and confirmations
+            var customFeeEstimation = new FeeEstimation
```

Comment:
why not just pass the `FeeEstimation` as a parameter instead of decimal and int?